cpp/celeborn/conf/BaseConf.cpp

```diff
@@ -0,0 +1,227 @@
+  folly::Optional<std::string> val;
+  auto it = lockedValues->find(key);
+  if (it != lockedValues->end()) {
+    val = it->second;
+  }
+  return val;
+}
+
+bool MemConfigMutable::isValueExists(const std::string& key) const {
+  auto lockedValues = values_.rlock();
+  return lockedValues->find(key) != lockedValues->end();
+}
+
+} // namespace core
+
+namespace util {
+
+std::unordered_map<std::string, std::string> readConfig(
+    const std::string& filePath) {
+  // https://teradata.github.io/presto/docs/141t/configuration/configuration.html
+
+  std::ifstream configFile(filePath);
+  if (!configFile.is_open()) {
+    CELEBORN_USER_FAIL("Couldn't open config file {} for reading.", filePath);
+  }
+
+  std::unordered_map<std::string, std::string> properties;
+  std::string line;
+  while (getline(configFile, line)) {
+    line.erase(std::remove_if(line.begin(), line.end(), isspace), line.end());
+    if (line[0] == '#' || line.empty()) {
+      continue;
+    }
+    auto delimiterPos = line.find('=');
+    auto name = line.substr(0, delimiterPos);
+    auto value = line.substr(delimiterPos + 1);
+    properties.emplace(name, value);
+  }
+
+  return properties;
+}
+
+std::string requiredProperty(
+    const std::unordered_map<std::string, std::string>& properties,
+    const std::string& name) {
+  auto it = properties.find(name);
+  if (it == properties.end()) {
+    CELEBORN_USER_FAIL("Missing configuration property {}", name);
+  }
+  return it->second;
+}
+
+std::string requiredProperty(
+    const Config& properties,
+    const std::string& name) {
+  auto value = properties.get(name);
+  if (!value.hasValue()) {
+    CELEBORN_USER_FAIL("Missing configuration property {}", name);
+  }
+  return value.value();
+}
+
+std::string getOptionalProperty(
+    const std::unordered_map<std::string, std::string>& properties,
+    const std::string& name,
+    const std::function<std::string()>& func) {
+  auto it = properties.find(name);
+  if (it == properties.end()) {
+    return func();
+  }
+  return it->second;
+}
+
+std::string getOptionalProperty(
+    const std::unordered_map<std::string, std::string>& properties,
+    const std::string& name,
+    const std::string& defaultValue) {
+  auto it = properties.find(name);
+  if (it == properties.end()) {
+    return defaultValue;
+  }
+  return it->second;
+}
+
+std::string getOptionalProperty(
+    const Config& properties,
+    const std::string& name,
+    const std::string& defaultValue) {
+  auto value = properties.get(name);
+  if (!value.hasValue()) {
+    return defaultValue;
+  }
+  return value.value();
+}
+
+} // namespace util
+
+BaseConf::BaseConf() : config_(std::make_unique<core::MemConfigMutable>()) {}
+
+void BaseConf::initialize(const std::string& filePath) {
+  // See if we want to create a mutable config.
+  auto values = util::readConfig(fs::path(filePath));
+  filePath_ = filePath;
+  checkRegisteredProperties(values);
+
+  bool mutableConfig{false};
```

Comment:
Maybe changing this config to true by default could be enough to align the default behavior of the Java side.